### PR TITLE
Change ABI suffix to Trait in dual case account and eth account traits

### LINKED ISSUE #1000: Change `ABI` suffix to `Trait` in dual case account and eth account traits
For naming consistency with the other dual case dispatcher traits

### PR DESCRIPTION
**Pull request by ggonzalez94**
Fixes #1000 

DON'T MERGE BEFORE `0.15.1` IS RELEASED

- [x] Added entry to CHANGELOG.md

### CODE REVIEW

**Comment by ericnordelo** [comment]
Thanks @ggonzalez94. This change while small is breaking the API, and we must add a Change (Breaking) entry to the changelog. Let's not merge it before 0.15.1 is released.

**Comment by ggonzalez94** [comment]
@ericnordelo just added the changelog. Agreed we shouldn't merge it before `0.15.1` is released.

File: CHANGELOG.md
Comment (ericnordelo) [inline CHANGELOG.md:14]:
```suggestion
- Changed ABI suffix to Trait in dual case account and eth account modules (#1096).
  - `DualCaseAccountABI` renamed to `DualCaseAccountTrait`
  - `DualCaseEthAccountABI` renamed to `DualCaseEthAccountTrait`
```

Comment (ericnordelo) [reply CHANGELOG.md:14]: These makes the changes easier to follow.

**Review by ericnordelo — COMMENTED** [review]
LGTM! Let's wait for 0.15.1 before merging it.

**Comment by ggonzalez94** [comment]
@ericnordelo with the release of `0.15.1` are we good to review and release this?

**Review by immrsd — APPROVED** [review]
LGTM!

File: CHANGELOG.md
Comment (ggonzalez94) [reply CHANGELOG.md]: It was my IDE that formatted this automatically. Just fixed it.

**Review by ericnordelo — APPROVED** [review]
LGTM!

**Pull request merged**
